marketplace/src/app/components/register/register.component.html

```diff
@@ -0,0 +1,49 @@
+<form [formGroup]="registerForm" (ngSubmit)="onRegister()">
+  <div class="grid">
+    <div class="col-12 lg:col-6">
+      <div class="card">
+        <h4>Personal details</h4>
+        <label for="firstName" class="block text-900 text-xl font-medium mb-2">First name</label>
+        <input id="firstName" type="text" placeholder="First name" pInputText class="w-full mb-3" formControlName="firstname" required minlength="2" autofocus>
+        <label for="familyName" class="block text-900 text-xl font-medium mb-2">Family name</label>
+        <input id="familyName" type="text" placeholder="Family name" pInputText class="w-full mb-3" formControlName="familyname" required minlength="2">
+        <label for="email2" class="block text-900 text-xl font-medium mb-2">Email</label>
+        <input id="email2" type="email" placeholder="Email address" pInputText class="w-full" formControlName="mail" required
+               (keydown.Tab)="getMailHint(registerForm.get('mail').value) && registerForm.get('mail').setValue(getMailHint(registerForm.get('mail').value))">
+        <p [hidden]="!getMailHint(registerForm.get('mail').value)">
+          Do you mean <a (click)="registerForm.get('mail').setValue(getMailHint(registerForm.get('mail').value))">{{getMailHint(registerForm.get('mail').value)}}</a>?
```

Comment:
It's better to use a button and style it so it looks like an "a" then using an "a" that functions as a button